src/components/AppointmentForm.tsx

```diff
@@ -1,255 +1,300 @@
 'use client'
 
-import { useTransition } from 'react'
+import { useTransition, useState } from 'react'
 import { useForm } from 'react-hook-form'
-import { appointmentFormAction } from './actions'
-import type AppointmentFormData from '@/types/AppointmentFormData'
+import { AppointmentFormData, Appointment } from '@/types/AppointmentFormData'
 import styles from '@styles/appointmentForm.module.css'
+import { createAppointment } from '@/utils/api'
+import AppointmentDetails from './AppointmentDetails'
+import Link from 'next/link'
 
 const AppointmentForm = () => {
   const [, startTransition] = useTransition()
+  const [createdAppointment, setCreatedAppointment] =
+    useState<Appointment | null>(null)
 
-  const defaultValues = {
-    firstName: '',
-    lastName: '',
-    phoneNum: '',
+  const defaultValues: AppointmentFormData = {
+    first_name: '',
+    last_name: '',
+    mobile_phone: '',
     email: '',
     dob: '',
-    apptDate: '',
-    apptTime: '',
-    apptType: '',
-    optionalDescription: '',
-    emergency: '',
+    requested_date: '',
+    requested_time: '',
+    appointment_type: '',
+    description: '',
+    is_emergency: '',
   }
 
   const {
     register,
     handleSubmit,
     formState: { errors, isValid, isDirty },
+    reset,
   } = useForm<AppointmentFormData>({
     defaultValues,
     mode: 'onChange',
   })
 
   const onSubmit = handleSubmit((data) => {
-    console.log(data)
     startTransition(() => {
-      appointmentFormAction(data)
+      createAppointment(data)
+        .then((createdAppointment) => {
+          console.log('Appointment created:', createdAppointment)
+          setCreatedAppointment(createdAppointment)
+        })
+        .catch((error) => {
+          console.error('Failed to create appointment:', error)
+          console.log(
+            'Appointment request not sent, please try again or call the phone number 123-123-1234',
+          )
+        })
     })
   })
```

Comment:
I'm just checking about this since I have not previously used useTransition with an actions.ts file, but is everything that is within this startTransition function supposed to live within the actions.ts file and the data should be sent to that with appointmentFormAction just as a way of slimming down the appointmentForm file? Again, I don't know if this is just a standard when using this method because really we don't need the actions file this way, right, and then we should just get rid of it? @5hraddha do you know?

Reply:
Thanks @Amanda2900 for pointing that as it completely skipped from my head. Yeah, the entire point of using `useTransition` hook was that we're using server actions to perform any server side data mutation. It could be any action that changes data at the server. 

In order to use server actions with supabase, we should still define the actions in `actions.ts` file & use it in the `useTransition` hook as we cannot invoke server action on form action (as we're using React Hook Form for the client side form validation).

@NuriyaAkh I know, it sounds a bit complicated. I found a video - [Progressively Enhanced Mutations with Next.js Server Actions](https://www.youtube.com/watch?v=Qc8_y9irMP4) that could help. We can go over this if we need more clarity in our call or sometime later one on one.